### PR TITLE
Add week 2 home work

### PR DESCRIPTION
**Pull request by leduyminh48**

### CODE REVIEW

File: homeworks/week2/Minh-Le.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    for (let i = 0; i < s.length; i++) {
+      if (!map[s[i]]) {
+        map[s[i]] = i // store index of first-occurence letters into the map
+      }
+    }
+
+    // remove duplicated character by indexes from the end
+    for (let i = s.length - 1; i >= 0; i--) {
+      if (i !== map[s[i]]) { // check if the current index is the first-occurence index of the letter
+        s.splice(i, 1);
```
Comment (hsn6) [inline homeworks/week2/Minh-Le.ts:28]: This line removes a character from the input string, is it O(1)? If not, what is the time complexity of this algorithm?
You should add some tests also. It's always a plus when candidate does that. It also tells us that you have tested your code.

Comment (leduyminh48) [reply homeworks/week2/Minh-Le.ts:28]: the splice method in JavaScript has time complexity of O(n), hence the time complexity here is more like O(2n) which is the same as O(n) I guess

Comment (hsn6) [reply homeworks/week2/Minh-Le.ts:28]: It's O(n^2). The easiest way to estimate is to count the number of nested loops. Another way is to write some examples and walk through it.

In this specific exercise, the number of unique characters is bounded so you can argue that O(n^2) will come back down to O(n) when string length is greater than a certain threshold. However, if this is not a string but an array of integer, the time complexity will stay at O(n^2).

Comment (leduyminh48) [reply homeworks/week2/Minh-Le.ts:28]: https://stackoverflow.com/questions/22614237/javascript-runtime-complexity-of-array-functions/22615787

**Pull request merged**